docs/knowledgebase/getting-started/glossary.md

```diff
@@ -220,80 +233,89 @@ some arguments. Such published functions take one additional parameter, known as
-is considered fundamentally Byzantine behaviour.
+A type of [Byzantine](#byzantine-fault-tolerance-bft) (erroneous/malicious) behavior that involves
+backing multiple mutually-exclusive options within the [consensus](#consensus) mechanism.
 
 ## Ethash
 
-One of a number of proofs of work, to be used in a Proof-of-Work (PoW) consensus algorithm.
-Originally developed for, and used in, Ethereum by a team led by Tim Hughes.
+A function used by some [proof-of-work](#proof-of-work) [consensus](#consensus) systems, notably
+that which is used by the Ethereum blockchain. It was developed by
+[a team led by Tim Hughes](https://github.com/ethereum/ethash/graphs/contributors).
 
 ## Events
 
-A means of recording, for the benefit of the off-chain world, that some particular state transition
-has happened. Within the context of the SRML, events are one of a number of amalgamatable datatypes
-that each module defines individually and which are aggregated together into a single overall `enum`
-type that can represent all module's types. Events are implemented through a transient set of
-storage items which are inspected immediately after a block has executed and reset during
-block-initialisation.
+A means of recording, for the benefit of the off-chain world, that some particular [state](#state)
+transition has happened. Within the context of [FRAME](#frame), events are one of a number of
+composable data types that each [pallet](#pallet) may individually define. Events in FRAME are
+implemented as a set of ephemeral storage items that are inspected immediately after a block has
+executed and reset during block-initialization.
 
 ## Executor
 
-A means of executing a function call in a given runtime with a set of externalities. There are two
-executor implementations present in Substrate, _Wasm_ and _Native_.
+A means of executing a function call in a given [runtime](#runtime) with a set of dependencies.
+There are two [executor](../../knowledgebase/advanced/executor) implementations present in
+Substrate, _Wasm_ and _Native_.
 
-### Wasm Executor
+### Native Executor
 
-An Executor that uses the Wasm binary and a Wasm interpreter to execute the needed call. This tends
-to be slow but is guaranteed correct.
+This executor uses a natively compiled runtime embedded in the [node](#node) to execute calls. This
+is a performance optimization that up-to-date nodes can take advantage of.
 
-### Native Executor
+### Wasm Executor
 
-An Executor that uses the current inbuilt and natively compiled runtime to execute the needed call.
-If the code is compatible with the on-chain Wasm code, then it will be a lot faster and correct.
-Incorrect versioning will result in a consensus error.
+This executor uses a [Wasm](#webassembly-wasm) binary and a Wasm interpreter to execute calls. Since
+Substrate-based chains include Wasm binaries as part of their [state](#state), this is guaranteed to
```

Comment:
"This" isn't clear - is it the binary or the executor that's up to date?
```suggestion
This executor uses a [Wasm](#webassembly-wasm) binary or a Wasm interpreter to execute calls. Since
Substrate-based chains include Wasm binaries as part of their [state](#state), the binary is guaranteed to
```